File: src/my_project/easy_problems/from201to250/element_appearing_more_than_25_percent.py
from typing import List 

class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        
        # initialize variables    
        len_arr: int = len(arr)
        temp_dict = dict()
        answer: int  = -1
        
        for i in range(len_arr):
            temp_dict[arr[i]] = 0
            
        for i in range(len_arr):
            temp_dict[arr[i]] += 1/len_arr
            
        for i in range(len_arr):
            if temp_dict[arr[i]] > 0.25:
                answer = arr[i]
                return answer 
        
        return answer 
    
    
solution = Solution()

print(solution.findSpecialInteger(arr = [1,2,2,6,6,6,6,7,10]))

print(solution.findSpecialInteger(arr = [1,1]))

File: src/my_project_typescript/easy_problems_typescript/from101to150/elementAppearingMoreThan25Percent.ts
function findSpecialInteger(arr: number[]): number {

    // initialize variables 
    let lenArr: number = arr.length
    let tempDict: any = {}
    let answer: number = -1

    // initialize the dictionary
    for (let i = 0; i < lenArr; i++){
        tempDict[arr[i]] = 0
    }

    // add the frequency in which each element occur
    for (let i = 0; i < lenArr; i++){
        tempDict[arr[i]] += 1/lenArr
    }

    // return the vale that occurs more than 25 percent of the time
    for (let i = 0; i < lenArr; i++){
        if (tempDict[arr[i]] > .25){
            answer = arr[i]
            return answer
        }
    }

    return answer 

};


console.log(findSpecialInteger([1,2,2,6,6,6,6,7,10]))

console.log(findSpecialInteger([1,1]))
